test/compound/TestSupplyVault.t.sol

```diff
@@ -394,8 +398,12 @@ contract TestSupplyVault is TestSetupVaults {
             daiSupplyVault.previewMint(amount),
             address(vaultSupplier1)
         );
-        (, uint128 userReward) = daiSupplyVault.userRewards(address(vaultSupplier1));
-        assertApproxEqAbs(uint256(userReward), expectedTotalRewardsAmount, 10000);
+        (, uint128 userReward1) = daiSupplyVault.userRewards(address(vaultSupplier1));
```

Comment:
In this we're not supplying on behalf. 

`index2` and `userReward2` should be > 0. This the opposite of withdraw on behalf